ngraph/test/runtime/interpreter/evaluates_map.cpp

```diff
@@ -1102,91 +1092,109 @@ namespace
-                break;
-            case element::Type_t::u16:
-                convert_v0::evaluate_bool<element::Type_t::u16>(op, outputs, inputs);
-                break;
-            case element::Type_t::u32:
-                convert_v0::evaluate_bool<element::Type_t::u32>(op, outputs, inputs);
-                break;
-            case element::Type_t::u64:
-                convert_v0::evaluate_bool<element::Type_t::u64>(op, outputs, inputs);
-                break;
-            case element::Type_t::f16:
-                convert_v0::evaluate_bool<element::Type_t::f16>(op, outputs, inputs);
-                break;
-            case element::Type_t::f32:
-                convert_v0::evaluate_bool<element::Type_t::f32>(op, outputs, inputs);
-                break;
-            case element::Type_t::f64:
-                convert_v0::evaluate_bool<element::Type_t::f64>(op, outputs, inputs);
-                break;
-            default: return false;
-            }
+        case element::Type_t::boolean:
+            convert_v0::evaluate<element::Type_t::boolean, OUT_ET>(op, outputs, inputs);
+            break;
+        case element::Type_t::i8:
+            convert_v0::evaluate<element::Type_t::i8, OUT_ET>(op, outputs, inputs);
+            break;
+        case element::Type_t::i16:
+            convert_v0::evaluate<element::Type_t::i16, OUT_ET>(op, outputs, inputs);
+            break;
+        case element::Type_t::i32:
+            convert_v0::evaluate<element::Type_t::i32, OUT_ET>(op, outputs, inputs);
+            break;
+        case element::Type_t::i64:
+            convert_v0::evaluate<element::Type_t::i64, OUT_ET>(op, outputs, inputs);
+            break;
+        case element::Type_t::u8:
+            convert_v0::evaluate<element::Type_t::u8, OUT_ET>(op, outputs, inputs);
+            break;
+        case element::Type_t::u16:
+            convert_v0::evaluate<element::Type_t::u16, OUT_ET>(op, outputs, inputs);
+            break;
+        case element::Type_t::u32:
+            convert_v0::evaluate<element::Type_t::u32, OUT_ET>(op, outputs, inputs);
+            break;
+        case element::Type_t::u64:
+            convert_v0::evaluate<element::Type_t::u64, OUT_ET>(op, outputs, inputs);
+            break;
+        case element::Type_t::f16:
+            convert_v0::evaluate<element::Type_t::f16, OUT_ET>(op, outputs, inputs);
+            break;
+        case element::Type_t::f32:
+            convert_v0::evaluate<element::Type_t::f32, OUT_ET>(op, outputs, inputs);
+            break;
+        case element::Type_t::f64:
+            convert_v0::evaluate<element::Type_t::f64, OUT_ET>(op, outputs, inputs);
+            break;
+        default: return false;
         }
-        else
+        return true;
+    }
+
+    namespace convert_like_v1
+    {
+        template <element::Type_t ti, element::Type_t to>
+        inline void evaluate(const shared_ptr<op::v1::ConvertLike>& op,
```

Comment:
I not sure but in my opinion `enable_if` should be on this level. `element_type_traits<T>::value_type` for `T == boolean` return `char` which is one of `unsigned char` or `signed char` and for one of for `T == i8` or `T == u8` will have the same `value_type` as `boolean`. 
I'm not sure but for now it looks like values for `i8` or for `u8` will be reduce to `boolean` [0,1].
Am I right?

Reply:
https://en.cppreference.com/w/cpp/language/types#Character_types